stan/math/prim/fun/mdivide_left_tri.hpp

```diff
@@ -122,16 +116,15 @@ inline Eigen::Matrix<double, R1, C2> mdivide_left_tri(
  *
  * @tparam TriView Specifies whether A is upper (Eigen::Upper)
  * or lower triangular (Eigen::Lower).
- * @tparam R1 number of rows, can be Eigen::Dynamic
- * @tparam C1 number of columns, can be Eigen::Dynamic
+ * @tparam T type of the matrix
  *
  * @param A Triangular matrix.
  * @return x = A^-1 .
  * @throws std::domain_error if A is not square
  */
-template <Eigen::UpLoType TriView, int R1, int C1>
-inline Eigen::Matrix<double, R1, C1> mdivide_left_tri(
-    const Eigen::Matrix<double, R1, C1> &A) {
+template <Eigen::UpLoType TriView, typename T, require_eigen_t<T> * = nullptr,
+          require_same_vt<double, T> * = nullptr>
```

Comment:
Could use ```require_eigen_vt``` here

Reply:
Only if we had `is_double` type trait. Maybe we should add it?

Reply:
We have require_floating_point and I'm pretty sure the OpenCL will case float to double if it sees a float

Reply:
> case float to double if it sees a float

I don't think so. It seem it would just copy raw data resulting in garbage output with no errors.